openmc/data/grid.py

```diff
@@ -0,0 +1,114 @@
+import numpy as np
+
+
+def linearize(x, f, tolerance=0.001):
+    """Return a tabulated representation of a function of one variable.
+
+    Parameters
+    ----------
+    x : Iterable of float
+        Initial x values at which the function should be evaluated
+    f : Callable
+        Function of a single variable
+    tolerance : float
+        Tolerance on the interpolation error
+
+    Returns
+    -------
+    numpy.ndarray
+        Tabulated values of the independent variable
+    numpy.ndarray
+        Tabulated values of the dependent variable
+
+    """
+    # Make sure x is a numpy array
+    x = np.asarray(x)
+
+    # Initialize output arrays
+    x_out = []
+    y_out = []
+
+    # Initialize stack
+    x_stack = [x[0]]
+    y_stack = [f(x[0])]
+
+    for i in range(x.shape[0] - 1):
+        x_stack.insert(0, x[i + 1])
+        y_stack.insert(0, f(x[i + 1]))
+
+        while True:
+            x_high, x_low = x_stack[-2:]
```

Comment:
While it may seem clear to the code author what this function is doing, it'd be great if there's a comment either explaining how the linearize operation is happening, or if a suitable link to the explanation for the calculations is provided.